packages/@aws-cdk/aws-iot-actions-alpha/lib/location-action.ts

```diff
@@ -0,0 +1,151 @@
+import { singletonActionRole } from './private/role';
+
+/**
+ * Location Timestamp Unit Enum
+ *
+ * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationTimestamp.html
+ */
+export enum LocationTimestampUnit {
+  /**
+   * SECONDS
+   */
+  SECONDS='SECONDS',
+
+  /**
+   * MILLISECONDS
+   */
+  MILLISECONDS='MILLISECONDS',
+
+  /**
+   * MICROSECONDS
+   */
+  MICROSECONDS='MICROSECONDS',
+
+  /**
+   * NANOSECONDS
+   */
+  NANOSECONDS='NANOSECONDS',
+}
+
+/**
+ * Describes how to interpret an application-defined timestamp value from an MQTT message payload and the precision of that value
+ *
+ * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationTimestamp.html
+ */
+export interface LocationTimestamp {
+  /**
+   * The precision of the timestamp value that results from the expression described in `value`.
+   * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationTimestamp.html
+   *
+   * @default MILLISECONDS
+   */
+  readonly unit?: LocationTimestampUnit
+
+  /**
+   * An expression that returns a long epoch time value
+   * @see https://docs.aws.amazon.com/iot/latest/developerguide/iot-substitution-templates.html
+   *
+   * @default '${timestamp()}'
+   */
+  readonly value?: string
+}
+
+/**
+ * Configuration properties of an action for Location.
+ */
+export interface LocationActionProps extends CommonActionProps {
+  /**
+   * The deviceId providing the location data.
+   * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationAction.html
+   */
+  readonly deviceId: string;
+
+  /**
+   * A string that evaluates to a double value that represents the latitude of the device's location
+   * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationAction.html
+   */
+  readonly latitude: string;
+
+  /**
+   * A string that evaluates to a double value that represents the longitude of the device's location.
+   * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationAction.html
+   */
+  readonly longitude: string;
+
+  /**
+   * The time that the location data was sampled. The default value is the time the MQTT message was processed..
+   * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationAction.html
+   *
+   * @default {unit: 'MILLISECONDS', value: '${timestamp()}'}
+   */
+  readonly timestamp?: LocationTimestamp;
+}
+
+/**
+ * The action to write the data from an MQTT message to an Amazon S3 bucket.
+ */
+export class LocationAction implements iot.IAction {
+  private readonly deviceId: string;
+  private readonly latitude: string;
+  private readonly longitude: string;
+  private readonly timestamp?: LocationTimestamp;
+  private readonly role?: iam.IRole;
+
+  /**
+   * @param tracker The Amazon Location tracker to which to write data.
+   * @param props Optional properties to not use default
+   */
+  constructor(private readonly tracker: location.CfnTracker, props: LocationActionProps) {
```

Comment:
We don't want to expose any L1s in an L2. For this PR we will need to implement an L2 for CfnTracker as well.

Reply:
I will try to add a L2 for the Location Tracker but will that be ok? The contribution guideline says that a new L2 must reference a RFC

Reply:
It looks like location tracker is a pretty simple resource and we already have an alpha module for location so we can skip the RFC.